codechain/run_node.rs

```diff
@@ -363,10 +363,11 @@ pub fn run_node(matches: &ArgMatches) -> Result<(), String> {
     }
 
     let _snapshot_service = {
+        let client = client.client();
+        let (tx, rx) = snapshot_notify::create();
+        client.engine().register_snapshot_notify_sender(tx);
```

Comment:
Why did you move this?

Reply:
It has a little bit of complicated story. In Tendermint, Worker waits some `register_*` functions to be called before it fires up its event loop. It is implemented with `crossbeam::channel` and it requires to be called orderly and thoroughly. So I moved it to there so the last `register_` function is called unconditionally.

Reply:
These weird initialization steps need to be cleaned up, but it should be an another PR.